nifi-mock/src/main/java/org/apache/nifi/util/MockFlowFile.java

```diff
@@ -290,4 +291,18 @@ public long getLineageStartIndex() {
     public long getQueueDateIndex() {
         return 0;
     }
+
+    public boolean isAttributeEqual(final String attributeName, final String expectedValue) {
+        // unknown attribute name, so cannot be equal.
+        if (attributes.containsKey(attributeName) == false)
+            return false;
+
+        String value = attributes.get(attributeName);
+        return Objects.equals(expectedValue, value);
+    }
+
+    public boolean isContentEqual(String excpected) {
+        final String value = new String(this.data, Charset.forName("UTF-8"));
+        return Objects.equals(excpected, value);
```

Comment:
Hi @JPercivall,

Thanks for reviewing.

Maybe I misunderstood, but assertContentEquals methods don’t return boolean and won’t work with Predicate’s.

For example, I can’t use assertContentEquals here (instead of isContentEqual)
```java
        runner.assertAllConditionsMet("success",
            mff -> mff.isAttributeEqual("GROUP_ATTRIBUTE_KEY", "1") && mff.isContentEqual("1,hello\n1,good-bye")
        );
```

isContentEqual parameter could be byte[] array.
But I think its convenient to have String parameter also, maybe 2 overloaded methods?

Thanks
Toivo